tracking/src/main/java/org/motechproject/nms/tracking/service/CacheService.java

```diff
@@ -0,0 +1,5 @@
+package org.motechproject.nms.tracking.service;
+
+public interface CacheService {
```

Comment:
It sounds like you could get access to the proxy and call through that:

```java
((TrackChangesService)AopContext.currentProxy())).isEntityInstance(arg)
```

Although that is frowned against since it couples you to the spring framework.  

I don't think we should be creating catchall CacheServices and moving methods that we want cached into them.  If we have to do it here then I'll grumpily accept, but we shouldn't do it in kilkari and all the other places we want to cache the results of a method.